phpBB/phpbb/storage/helper.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\storage;
+
+use phpbb\config\config;
+use phpbb\di\service_collection;
+
+class helper
+{
+	/** @var config */
+	protected $config;
+
+	/** @var service_collection */
+	protected $provider_collection;
+
+	/** @var service_collection */
+	protected $adapter_collection;
+
+	/** @var adapter_factory */
+	protected $adapter_factory;
+
+	/** @var state_helper */
+	protected $state_helper;
+
+	public function __construct(config $config, service_collection $provider_collection, service_collection $adapter_collection, adapter_factory $adapter_factory, state_helper $state_helper)
+	{
+		$this->config = $config;
+		$this->provider_collection = $provider_collection;
+		$this->adapter_collection = $adapter_collection;
+		$this->adapter_factory = $adapter_factory;
+		$this->state_helper = $state_helper;
+	}
+
+	/**
+	 * Get adapter definitions from a provider
+	 *
+	 * @param string $provider Provider class
+	 * @return array Adapter definitions
+	 */
+	public function get_provider_options(string $provider) : array
+	{
+		return $this->provider_collection->get_by_class($provider)->get_options();
+	}
+
+	/**
+	 * Get the current provider from config
+	 *
+	 * @param string $storage_name Storage name
+	 * @return string The current provider
+	 */
+	public function get_current_provider(string $storage_name) : string
+	{
+		return (string) $this->config['storage\\' . $storage_name . '\\provider'];
+	}
+
+	/**
+	 * Get the current value of the definition of a storage from config
+	 *
+	 * @param string $storage_name Storage name
+	 * @param string $definition Definition
+	 * @return string Definition value
+	 */
+	public function get_current_definition(string $storage_name, string $definition) : string
+	{
+		return (string) $this->config['storage\\' . $storage_name . '\\config\\' . $definition];
+	}
+
+	/**
+	 * Get current storage adapter
+	 *
+	 * @param string $storage_name Storage adapter name
+	 *
+	 * @return object Storage adapter instance
+	 */
+	public function get_current_adapter(string $storage_name): object
+	{
+		static $adapters = [];
+
+		if (!isset($adapters[$storage_name]))
+		{
+			$adapters[$storage_name] = $this->adapter_factory->get($storage_name);
+		}
+
+		return $adapters[$storage_name];
+	}
+
+	/**
+	 * Get new storage adapter
+	 *
+	 * @param string $storage_name
+	 *
+	 * @return mixed Storage adapter instance
```

Comment:
Isn't there an interface these are following?

Reply:
Yes, but the class extends from 2 interfaces, so it cant be typed